src/inc/RingBuffer.h

```diff
@@ -0,0 +1,149 @@
+#if !defined(__RingBuffer_hdr__)
+#define __RingBuffer_hdr__
+
+#include <cassert>
+#include <algorithm>
+#include <cmath>
+#include <cstring>
+
+/*! \brief implement a circular buffer of type T
+*/
+template <class T>
+class CRingBuffer
+{
+public:
+    explicit CRingBuffer(int iBufferLengthInSamples) :
+        m_iBuffLength(iBufferLengthInSamples),
+        m_iReadIdx(0),
+        m_iWriteIdx(0),
+        m_ptBuff(0)
+    {
+        assert(iBufferLengthInSamples > 0);
```

Comment:
Not sure what happened, but I got my testing break here. It's working again after commented this line.
Error Message:
`(base) shanjiang@lawn-143-215-124-120 debug % ./musi6106exec /Users/shanjiang/documents/github/2022-musi6106/bin/debug/sweep.wav fir 0.001 1`
`MUSI6106 Assignment Executable`
`(c) 2014-2022 by Alexander Lerch`

`Assertion failed: (iBufferLengthInSamples > 0), function CRingBuffer, file RingBuffer.h, line 21.`
`zsh: abort      ./musi6106exec  fir 0.001 1`